### PR TITLE
logger: use lazy formatting

### LINKED ISSUE #1843: logger: use logging format interpolation
E.g.

```
logger.debug("Saving {} to {}.".format(path_info, to_info))
```
should be
```
logger.debug("Saving '%s' to '%s'.", path_info, to_info)
```

~~As suggested by @ei-grad in this PR: https://github.com/iterative/dvc/pull/1822~~

~~Reference: https://docs.python.org/3/howto/logging.html#optimization~~

~~It needs to support braces `{}` formatting, you can use a log adapter for Python 2.7:~~
<s>```python
import logging

class Message(object):
    def __init__(self, fmt, args):
        self.fmt = fmt
        self.args = args

    def __str__(self):
        return self.fmt.format(*self.args)

class StyleAdapter(logging.LoggerAdapter):
    def __init__(self, logger, extra=None):
        super(StyleAdapter, self).__init__(logger, extra or {})

    def log(self, level, msg, *args, **kwargs):
        if self.isEnabledFor(level):
            msg, kwargs = self.process(msg, kwargs)
            self.logger._log(level, Message(msg, args), (), **kwargs)
```</s>

### PR DESCRIPTION
**Pull request by alexmojaki**
Fixes #1843

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

### CODE REVIEW

File: dvc/analytics.py [hunk 1]
```diff
@@ -187,4 +187,4 @@ def _find_or_create_user_id():
             return user_id
 
     except LockError:
-        logger.debug(f"Failed to acquire '{lockfile}'")
+        logger.debug("Failed to acquire %r", lockfile)
```
Comment (alexmojaki) [inline dvc/analytics.py:190]: Here I have changed `'{lockfile}'` to `%r` (i.e. use the `repr`, as opposed to `%s`) which takes care of quotes and looks better if `lockfile` itself happens to contain single quotes. I've done this several times in this PR wherever the value appears to be a string.

File: dvc/repo/experiments/push.py [hunk 2]
```diff
@@ -85,7 +85,7 @@ def _push(
     from ...scm import GitAuthError
 
     refspec_list = [f"{exp_ref}:{exp_ref}" for exp_ref in refs]
-    logger.debug(f"git push experiment '{refs}' -> '{git_remote}'")
+    logger.debug("git push experiment %r -> '%s'", refspec_list, git_remote)
```
Comment (alexmojaki) [inline dvc/repo/experiments/push.py:88]: Here I've changed `refs` to `refspec_list` to match `pull.py` just above.

File: dvc/analytics.py (hunk 1 above)
Comment (alexmojaki) [reply dvc/analytics.py:190]: Oops, this caused tests to fail on Windows with good reason, as backslashes got escaped too, turning `'C:\\Users\\runneradmin\\AppData...'` into `'C:\\\\Users\\\\runneradmin\\\\AppData...`. So I guess I'll revert all of these to just `'%s'`. Any objections?

File: dvc/data_cloud.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def _init_odb(self, name):
         return get_odb(cls(**config), fs_path, **config)
 
     def _log_missing(self, status: "CompareStatusResult"):
-        if status.missing:
+        if status.missing and logger.isEnabledFor(logging.WARNING):
```
Comment (efiop) [inline dvc/data_cloud.py:67]: hm, could you explain the reasoning behind this change, please? Seems unrelated to the rest.

**Review by efiop — APPROVED** [review]
Thank you!

File: dvc/data_cloud.py (hunk 3 above)
Comment (efiop) [reply dvc/data_cloud.py:67]: @alexmojaki That seems unnecessary though. I'll adjust it back for now
